src/lib.rs

```diff
@@ -98,21 +108,26 @@ where
     /// # use libwallet::{Wallet, SimpleVault, sr25519, Result};
     /// # #[async_std::main] async fn main() -> Result<()> {
     ///
-    /// let wallet = Wallet::new(SimpleVault::<sr25519::Pair>::new()).unlock("").await?;
+    /// let wallet = Wallet::new(SimpleVault::<sr25519::Pair>::new()).unlock(()).await?;
     /// let signature = wallet.sign(&[0x01, 0x02, 0x03]);
     /// assert!(signature.is_ok());
     /// # Ok(()) }
     /// ```
     pub fn sign(&self, message: &[u8]) -> Result<SignatureOf<V, C>> {
-        Ok(self.root_account()?.sign(message))
+        match self.root_account()?.sign(message) {
+            Ok(signature) => Ok(signature),
+            Err(_) => Err(Error::Locked),
+        }
+        //Ok()
     }
 
     /// Switch the network used by the root account which is used by
     /// default when deriving new sub-accounts
-    pub fn switch_default_network(&mut self, net: &str) -> Result<&Account<V::Pair>> {
+    pub fn switch_default_network(&mut self, net: &str) -> Result<Rc<Account<V::Pair>>> {
         let root = self.root.take();
         let network = net.parse().map_err(|_| Error::InvalidNetwork)?;
-        self.root = root.map(|a| a.switch_network(network));
+        self.root = root.map(|a| Rc::new(Rc::try_unwrap(a).ok().unwrap().switch_network(network)));
```

Comment:
I think this was already changed, but as you said, I should handle the error.

Reply:
The thing with this `try_unwrap` is that as soon as we clone the Rc somewhere this is going to fail(now worse because it panics) which is going to be most of the time since people are encouraged to use sub-accounts instead of the root account directly. Since we want to change the `root` calling the `switch_network` that consumes/moves the type, the usual way to proceed it's to use an `Cell` that wraps our type(e.g. `Rc<Cell<Account>>`) to give us (interior)mutability, `RefCell` would be used if we want references or we could also make the whole `switch_default_network` method also consume `self` but then we should probably return the wallet instead of the account